### PR TITLE
Draft Minutes for HSF Packaging Meeting 24

### PR DESCRIPTION
**Pull request by drbenmorgan**
Collated from live notes and links added. I think all the names are complete and correct!

### CODE REVIEW

File: organization/_posts/2018-05-16-packaging.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- Graeme: Overview on [Guix Cern Computing Seminar](https://indico.cern.ch/event/719851/) (pronounced "gix"!)
+  - No slides, but link to recording available
+  - Nutshell: very similar to nix in how packages built, hashed.
+  - High reproducibility (up to 95% bitwise identical).
+  - Nix-style runtime env setup
+  - Everything lives under `/guix` (must be writable, user builds supported by daemon). Binary builds supported, but not relocatable (prefix enters package hash).
+  - User Base: mostly HPC, big shared filesystems user are close to (Graeme commented at seminar that typical HEP case is different; CVMFS etc)
+  - Good points like Nix: reproducibility, user control, composability (overlays)
+  - Differences: Scheme language vs Nix DSL (claimed Scheme makes use/extension easier). No mention of license
+  - Interesting for us? Yes, but lives very close to Nix. Limited resources, so not a priority unless someone wants to take it on. Maybe more important if we move in a Nix-like direction. Then differences between Nix DSL vs Scheme etc.
```
Comment (graeme-a-stewart) [inline organization/_posts/2018-05-16-packaging.md]: Slightly odd last sentence. Better would be "Then differences between Nix DSL vs Scheme, for example, would be more relevant to investigate."

File: organization/_posts/2018-05-16-packaging.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - Overall README and a template for each specific tool
+  - Walk through the minimum packages and versions
+  - Demonstrate basic functionality first
+  - Template can be filled in so that each package manager is well described and easy to use by the non-expert test driver
+- Graeme: time limited on progress on Nix due to meetings and travel.
+- Chris Burr: With Nix, now packaging LHCb up to reconstruction packages.
+- Javier - has discussed with Patrick which repo to use for package descriptions. HSF looks the best. CERN one is very out of date.
+- Patrick has built most of the CMSSW stack, most updates went into HSF. Chris has been adding more versions, PRs going directly upstream into Spack. There is a FNAL fork that has useful changes to Spack itself and to some of the packages.
+  - Could add that information to packaging github repo, so that people could find it.
+  - Shazad has been working on a tool to convert SCRAM->CMake configurations, but it was taking too long for CMS (10 minutes - too slow!). CMake 3.11 is improved. Ninja target was faster than make. Some technical improvements being worked on.
```
Comment (graeme-a-stewart) [inline organization/_posts/2018-05-16-packaging.md]: Suggest: "Shazad has been working on a tool to convert SCRAM->CMake configurations. However, resulting CMake configuration step was taking too long for CMS (10 minutes). CMake 3.11 improved the situation. Ninja target was faster than make. Some technical improvements being worked on."

File: organization/_posts/2018-05-16-packaging.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- Ben: Giulio has submitted a PR for an [aliBuild testdrive](https://github.com/HSF/packaging/pull/12). Comments/Review by all welcome!
+- Guilherme - wants to make a special container for the Portage. Will put information on the website.
+
+AOB
+===
+- Due to the [LIM workshop on 30 May](https://indico.cern.ch/event/720948/) the next meeting will be on the [13th June](https://indico.cern.ch/event/730538/)
+- Noted the "HSF" user on DockerHub : Ask Graeme if you want to be added here to add images for test driving, etc.
```
Comment (graeme-a-stewart) [inline organization/_posts/2018-05-16-packaging.md]: HSF -> hepsoftwarefoundation

Link is worth adding: https://hub.docker.com/u/hepsoftwarefoundation/dashboard/

**Comment by drbenmorgan** [comment]
Thanks @graeme-a-stewart, fixed, only minor thing was the link to Docker was 404'ing, so changed to "https://hub.docker.com/u/hepsoftwarefoundation/". I think the review outcome was o.k (requested changes?).

**Comment by graeme-a-stewart** [comment]
OK, that link must only work if you are a member of the organisation - good thing you checked!

**Pull request merged**